code/modules/flufftext/Hallucination.dm

```diff
@@ -702,307 +739,337 @@ Gunshots/explosions/opening doors/less rare audio (done)
+		//The truth, revealed
+		"<span class='warning'>You're hallucinating.</span>", \
+		//Direct speech
+		"<span class='notice'>[pick("Hmm...","Yes.","No.","Ignore that last message.","You should stop doing that.","Good luck.")]</span>")
 	to_chat(target, chosen)
 	qdel(src)
 
-/mob/living/carbon/proc/hallucinate(hal_type, specific) // specific is used to specify a particular hallucination
-	set waitfor = 0
-	switch(hal_type)
-		if("xeno")
-			new /obj/effect/hallucination/xeno_attack(src.loc,src)
-		if("singulo")
-			new /obj/effect/hallucination/singularity_scare(src.loc,src)
-		if("koolaid")
-			new /obj/effect/hallucination/oh_yeah(src.loc,src)
-		if("battle")
-			new /obj/effect/hallucination/battle(src.loc,src)
-		if("flood")
-			new /obj/effect/hallucination/fake_flood(src.loc,src)
-		if("delusion")
-			new /obj/effect/hallucination/delusion(src.loc,src)
-		if("self_delusion")
-			new /obj/effect/hallucination/self_delusion(src.loc,src)
-		if("fake")
-			new /obj/effect/hallucination/fakeattacker(src.loc,src)
-		if("bolts")
-			new /obj/effect/hallucination/bolts(src.loc,src)
-		if("bolts_minor")
-			new /obj/effect/hallucination/bolts(src.loc,src,rand(1,2))
-		if("whispers")
-			new /obj/effect/hallucination/whispers(src.loc,src)
-		if("message")
-			new /obj/effect/hallucination/message(src.loc,src)
-		if("items_other")
-			new /obj/effect/hallucination/items_other(src.loc,src)
-		if("sounds")
-			//Strange audio
-			switch(rand(1,20))
-				if(1) playsound_local(null,'sound/machines/airlock.ogg', 15, 1)
+/datum/hallucination/sound
+	cost = 15
+
+/datum/hallucination/sound/New(mob/living/carbon/T, forced = TRUE)
+	..()
+	//Strange audio
+	switch(rand(1,20))
+		if(1)
+			target.playsound_local(null,'sound/machines/airlock.ogg', 15, 1)
+		if(2)
+			if(prob(50))
+				target.playsound_local(null,'sound/effects/explosion1.ogg', 50, 1)
+			else
+				target.playsound_local(null, 'sound/effects/explosion2.ogg', 50, 1)
+		if(3)
+			target.playsound_local(null, 'sound/effects/explosionfar.ogg', 50, 1)
+		if(4)
+			target.playsound_local(null, pick('sound/effects/glassbr1.ogg','sound/effects/glassbr2.ogg','sound/effects/glassbr3.ogg'), 50, 1)
+		if(5)
+			target.playsound_local(null, 'sound/weapons/ring.ogg', 35)
+			sleep(15)
+			target.playsound_local(null, 'sound/weapons/ring.ogg', 35)
+			sleep(15)
+			target.playsound_local(null, 'sound/weapons/ring.ogg', 35)
+		if(6)
+			target.playsound_local(null, 'sound/magic/summon_guns.ogg', 50, 1)
+		if(7)
+			target.playsound_local(null, 'sound/machines/alarm.ogg', 100, 0)
+		if(8)
+			target.playsound_local(null, 'sound/voice/bfreeze.ogg', 35, 0)
+		if(9)
+			target.playsound_local(null, 'sound/effects/pray_chaplain.ogg', 50)
+	//Rare audio
+		if(10)
+	//These sounds are (mostly) taken from Hidden: Source
+			var/list/creepyasssounds = list('sound/effects/ghost.ogg', 'sound/effects/ghost2.ogg', 'sound/effects/heart_beat.ogg', 'sound/effects/screech.ogg',\
```

Comment:
We can start enforcing sort of a naming convention. Like adding the context to the list, so this would be `hallucinations_creepyasssounds`